### PR TITLE
feat(rwa): add T-REX compliance modules for RWA token stack

### PR DESCRIPTION
**Pull request by pasevin**
## Summary

Ports 7 T-REX (ERC-3643) compliance modules from EVM/Solidity to Soroban/Rust, targeting the closest possible 1:1 parity with the [Tokeny T-REX reference implementation](https://github.com/TokenySolutions/T-REX/tree/main/contracts/compliance/modular/modules). Each module is a standalone deployable contract under `packages/compliance-modules/`.

**Modules:**
| Module | T-REX Equivalent | Purpose |
|---|---|---|
| `CountryAllowModule` | `CountryAllowModule` | Allowlist of permitted investor countries |
| `CountryRestrictModule` | `CountryRestrictModule` | Blocklist of restricted investor countries |
| `MaxBalanceModule` | `MaxBalanceModule` | Per-identity balance cap |
| `SupplyLimitModule` | `SupplyLimitModule` | Global token supply ceiling |
| `TransferRestrictModule` | `TransferRestrictModule` | Per-address transfer allowlist |
| `TimeTransfersLimitsModule` | `TimeTransfersLimitsModule` | Rate-limited transfers per identity per time window |
| `InitialLockupPeriodModule` | `TimeExchangeLimitsModule` | Token lockup period after mint |

**Key architectural decisions:**
- **Internal state tracking** — Soroban forbids contract re-entry, so modules that need token state (`SupplyLimitModule`, `InitialLockupPeriodModule`) maintain internal mirrors updated via `on_*` hooks, instead of calling back to the token. This follows the same proven pattern used by `MaxBalanceModule`. See [architecture document](https://www.notion.so/openzeppelin/Soroban-Re-Entry-Limitation-EVM-vs-Stellar-Compliance-Architecture-319cbd12786081bfa4bff8b625d957d7).
- **Hook wiring verification (arming)** — 4 stateful modules expose `required_hooks()` and `verify_hook_wiring()` methods. After deployment and wiring, calling `verify_hook_wiring()` cross-calls the compliance contract to confirm all required hooks are registered. `can_*` hooks panic with a descriptive message if the module is not armed, preventing silent misconfiguration.
- **Standalone crates** — Each module lives in its own crate under `packages/compliance-modules/modules/`, with shared helpers in `packages/compliance-modules/common/`.

**What's included:**
- 7 compliance module crates + 1 shared common crate
- `examples/rwa-compliance/` — full-stack integration example wiring a real RWA token through a compliance dispatcher to all 7 modules, with **23 passing tests** covering happy paths, edge cases, burn flows, time window resets, country-based rejections, and wiring guard enforcement
- `packages/compliance-modules/scripts/` — build, deploy, wire, and e2e test scripts for Stellar testnet (all **17 on-chain assertions pass**)
- `packages/compliance-modules/deploy/` — lightweight infrastructure contracts (IRS, verifier, compliance, token) for testnet deployment
- Documentation: `packages/compliance-modules/README.md` with hook wiring matrix, arming mechanism docs, and script usage

## Test plan
- [x] `cargo test -p rwa-compliance-example` — 23/23 tests pass
- [x] Full e2e testnet deployment (`scripts/e2e.sh`) — 11 WASMs built, 11 contracts deployed, 19 hooks wired, 4 modules verified, 3 investors registered, 17/17 on-chain assertions pass
- [x] Verified `CountryAllowModule` blocks mint to non-allowed country (Japan/392)
- [x] Verified `CountryRestrictModule` blocks mint to restricted country (DPRK/408)
- [x] Verified `SupplyLimitModule` rejects over-mint at 10M ceiling
- [x] Verified `MaxBalanceModule` rejects mint above 1M per-identity cap
- [x] Verified `InitialLockupPeriodModule` blocks transfer and burn during lockup
- [x] Verified internal supply counter tracks correctly across multiple mints
- [x] Wiring guard tests confirm `#[should_panic]` for unarmed modules and missing hooks

## Summary by CodeRabbit

* **New Features**
  * Added RWA (Real World Asset) compliance framework with modular architecture
  * Introduced seven compliance modules: country allowlist/restrictions, max balance, supply limits, time-based transfer limits, transfer restrictions, and initial lockup periods
  * Added identity registry and verification system for managing investor identities and compliance
  * Implemented RWA token contract with admin controls, freezing, and compliance integration

* **Tests**
  * Added comprehensive test suite validating compliance module interactions and workflows

* **Chores**
  * Added build, deployment, and testing automation scripts for compliance infrastructure

### CODE REVIEW

File: packages/compliance-modules/modules/max-balance/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        let to_id = irs.stored_identity(&to);
+
+        let to_balance: i128 = e
+            .storage()
+            .persistent()
+            .get(&DataKey::IDBalance(token, to_id))
+            .unwrap_or_default();
+
+        checked_add_i128(e, to_balance, amount) <= max
+    }
```
[the remark this answers is not in the document]
Comment (pasevin) [reply packages/compliance-modules/modules/max-balance/src/lib.rs]: Fixed in 6bf197c — `can_transfer` now resolves both `from` and `to` identities and returns `true` early when `from_id == to_id`, matching the existing short-circuit in `on_transfer`.

**Comment by ozgunozerk** [comment]
This turned out to be a very big PR, and hard to review since there will be many iterations. 

If you want to keep it as a one big PR, it is fine by me, but then here is how it will work out from my end: 
- I'll only review the high-level design at first, and won't dive into the details of the modules.
- after architectural/design decisions are finalized (which may take couple iterations), only then we can start addressing smaller details
- diving into modules themselves, and finalizing the nitpicks, may also take couple of iterations

If you want to have fewer rounds of iterations, and merge faster, I suggest dividing this PR into multiple ones. For example:
- each module gets their own PR (as stated in the cursor plan)
- the first PR will be the architecture of the modules, and the `common` file

I'm just laying out the potential workflow we will have here to set expectations to prevent anyone getting discouraged ❤️
